### PR TITLE
pyenv related docs added, warning message in breeze initialize-local-virtualenv command

### PR DESCRIPTION
**Pull request by Bowrna**
Pyenv related docs - macOS Big Sur pyenv issue can be fixed by adding commands into ~/.zshrc file
Having AIRFLOW_HOME and Source code in ${HOME} can lead to disastrous effect of deleting all the file in the Airflow folder. Added a warning message to prevent deletion

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: breeze [hunk 1]
```diff
@@ -283,6 +283,13 @@ function breeze::initialize_virtualenv() {
         echo
         echo "Wiping and recreating ${AIRFLOW_HOME_DIR}"
         echo
+        SCRIPTPATH="$( cd -- "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 ; pwd -P )"
+        if [ "${SCRIPTPATH}" = "${AIRFLOW_HOME_DIR}" ]; then
+            echo "AIRFLOW_HOME and Source code for Apache Airflow resides in the same path ${AIRFLOW_HOME_DIR}"
+            echo "When running this command it will delete all the files in the path ${AIRFLOW_HOME_DIR} to clear dynamic files like config/logs/db"
+            echo "Move your source code for Apache Airflow to different folder to avoid deletion"
+            exit 0
```
Comment (potiuk) [inline breeze]: Exit should be with error code (exit 1 is fine)

File: breeze [hunk 2]
```diff
@@ -283,6 +283,13 @@ function breeze::initialize_virtualenv() {
         echo
         echo "Wiping and recreating ${AIRFLOW_HOME_DIR}"
         echo
+        SCRIPTPATH="$( cd -- "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 ; pwd -P )"
+        if [ "${SCRIPTPATH}" = "${AIRFLOW_HOME_DIR}" ]; then
```
Comment (potiuk) [inline breeze]: We prefer [[ ]] over [] - generally we try to follow Google she guide - you can read more on why [[]] is better  https://google.github.io/styleguide/shellguide.html#test----and---

File: breeze [hunk 3]
```diff
@@ -283,6 +283,13 @@ function breeze::initialize_virtualenv() {
         echo
         echo "Wiping and recreating ${AIRFLOW_HOME_DIR}"
         echo
+        SCRIPTPATH="$( cd -- "$(dirname "${BASH_SOURCE[0]}")" >/dev/null 2>&1 ; pwd -P )"
```
Comment (potiuk) [inline breeze]: We already have AIRFLOW_SOURCES that we can use here (see beginning of the breeze script).

**Review by potiuk — CHANGES_REQUESTED** [review]
Few small things to improve

File: breeze [hunk 4]
```diff
@@ -283,6 +283,12 @@ function breeze::initialize_virtualenv() {
         echo
         echo "Wiping and recreating ${AIRFLOW_HOME_DIR}"
         echo
+        if [[ "${AIRFLOW_SOURCES}" = "${AIRFLOW_HOME_DIR}" ]]; then
```
Comment (potiuk) [inline breeze]: I think you need == here otherwise it is assignment not comparision

**Review by potiuk — CHANGES_REQUESTED** [review]

**Comment by Bowrna** [comment]
@potiuk One of the static code checks is the spelling error. I have given the macOS version name as `Big Sur`, but the check wants me to fix the spelling for the below line that i have added
`4. If you are in macOS Big Sur 11.4, pyenv may face issues. It can be fixed by adding the below command to ~/.zshrc`
```
Run codespell to check for common misspellings in files..................................Failed
- hook id: codespell
- exit code: 65

CONTRIBUTORS_QUICK_START.rst:150: Sur ==> Sure, sir
```
How could I handle this error? Can you guide me on this?

**Comment by ephraimbuddy** [comment]
> While i am usually tempted to do the same 'yeah small change let's do it together' - learned by experience it might bite back and when i feel i need to add ' ... and this change also does this ...' in commit message it is usually a sign we should split.
> 
> There is very little penalty of splitting commits/PR into separate ones.
> 
> And yes i often fail myself to follow this advice. i am.ad guilty as anyone of having more than one change In a logical commit. But i do my best to improve :)

Thanks very much, @potiuk. That's very good advice. 

Please @Bowrna make a separate PR for this

**Comment by Bowrna** [comment]
Got it. let me fix it and raise a different PR. Do I need to make any changes to this PR further? If yes, please let me know. Thanks for all the clarification @ephraimbuddy @potiuk

**Review by potiuk — APPROVED** [review]

**Pull request merged**